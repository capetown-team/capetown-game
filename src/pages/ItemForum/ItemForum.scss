@import '../../styles/layout/colors';
@import '../../styles/layout/variables';

.item-forum {
  min-height: calc(100vh - #{$headerHeight});
  display: flex;
  flex: 1 1 auto;
  justify-content: space-between;
  flex-direction: column;

  &__comments {
    flex: 1 1 auto;
  }
}
