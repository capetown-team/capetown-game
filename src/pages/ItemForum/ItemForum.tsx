import React, { memo, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import block from 'bem-cn-lite';

import { Topping } from '@/components/Topping';
import { PageMeta } from '@/components/PageMeta';
import { ForumComment } from '@/pages/ItemForum/ForumComment';
import { Editor } from '@/components/Editor';
import { Loading } from '@/components/Loading';

import { commentsSelector, pendingSelector } from '@/reducers/forum/selectors';
import { getComments } from '@/reducers/forum/actions';
import { getTopicId } from '@/pages/Forum/helper';

import { AppState } from '@/reducers';

import { messageProps } from '@/reducers/forum/types';

import './ItemForum.scss';

const b = block('item-forum');

const ItemForum = () => {
  const dispatch = useDispatch();
  const topicId = getTopicId();
  const [data, setData] = useState([]);

  useEffect(() => {
    console.log('topicId', topicId);
    dispatch(getComments(topicId));
  }, [dispatch]);

  const { currentData, loading } = useSelector((state: AppState) => {
    console.log('data', data);
    return {
      currentData: commentsSelector(state),
      loading: pendingSelector(state)
    };
  });
  useEffect(() => {
    console.log('curdata', currentData[0]);
    if (currentData !== undefined) {
      setData(currentData[0]);
    }
  }, [currentData]);

  if (loading) {
    return <Loading />;
  }

  return (
    <div className={b()}>
      <PageMeta title="Форум" description="Какая интересная игра!" />
      <Topping title="Делимся секретами игры" />

      <div className={b('comments')}>
        {data.map((item: messageProps) => (
          <ForumComment
            right={false}
            name={item.name}
            content={item.content}
            time={item.time}
            countComments={item.countComments}
            countLikes={item.countLikes}
          />
        ))}
      </div>

      <Editor small={false} />
    </div>
  );
};
const WrappedItemForum = memo(ItemForum);

export { WrappedItemForum as ItemForum };
