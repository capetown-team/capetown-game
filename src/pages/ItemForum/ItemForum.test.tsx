import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { ItemForum } from './ItemForum';

describe('<Authorization />', () => {
  it('should renders correct <ItemForum />', () => {
    const wrapper = shallow(<ItemForum />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
