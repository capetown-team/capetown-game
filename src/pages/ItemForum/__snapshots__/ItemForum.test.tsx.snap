// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authorization /> should renders correct <ItemForum /> 1`] = `
<Fragment>
  <Memo(Topping)
    title="Форум"
  />
  <hr />
  <p>
    Какая интересная игра!
  </p>
</Fragment>
`;
